app/react-native/src/preview/components/OnDeviceUI/index.js

```diff
@@ -1,21 +1,54 @@
 import React, { Component, PropTypes } from 'react';
-import { View, Text } from 'react-native';
+import { View, Text, SegmentedControlIOS } from 'react-native';
 import style from './style';
 import StoryListView from '../StoryListView';
 import StoryView from '../StoryView';
 
 export default class OnDeviceUI extends Component {
+  constructor(props) {
+    super(props)
+    this.screenSizes = {
+      'iPhone 4': { width: 320, height: 480 },
+      'iPhone SE': { width: 320, height: 568 },
+      'iPhone 7': { width: 375, height: 667 },
+      'iPhone 7 Plus': { width: 414, height: 736 },
+    };
+    this.screenSizeNames = Object.keys(this.screenSizes);
+
+    this.state = {
+      selectedSizeIndex: 0,
+    };
+  }
+
   render() {
     const {
       stories,
       events,
       url
     } = this.props;
 
+    const canvasSize = this.screenSizes[this.screenSizeNames[this.state.selectedSizeIndex]];
+
     return (
       <View style={style.main}>
         <StoryListView stories={stories} events={events} />
-        <StoryView url={url} events={events} />
+        <View style={style.preview}>
+          <SegmentedControlIOS
```

Comment:
I've just added a screenshot, but I'll add a gif when it's a bit more thought through.

https://github.com/storybooks/storybook/pull/1413#issuecomment-313426921